spec/models/batch_job_spec.rb

```diff
@@ -85,8 +85,9 @@
 
   describe "#check_complete" do
     it "when destroyed by another checker" do
-      job = described_class.create!.tap(&:destroy)
+      job = described_class.new
```

Comment:
How is this related to rspec mocks?

Reply:
Because a destroyed object is frozen, and rspec 3.10 raises an error about mocking a frozen object (see the error referenced here https://github.com/ManageIQ/miq_bot/pull/566#issuecomment-904082377)

Reply:
So, should we set up the expectation before destroying the record instead?

Reply:
Sure